### DISCUSSION TITLE
Keep statistics after restart

### DISCUSSION THREAD

**Discussion by sorooshab** [v2fly/v2ray-core#2290] [Q&A]
I have some ways to get statistics using api stats and update it to the database, however after server restart, stats data restart and mess up the data. is it possible to keep the stats from being reset or any way to manually modify the stats data? so that after restart I can get the previous data and update it in v2ray.

**Comment by mydogshitgold** [comment]
I don't think you can preserve statistics, which are stored in the memory only, after restart.
afaik there isn't a [grpc service](https://github.com/v2fly/v2ray-core/blob/75addbf19528164155b89c823efc94ac6219b958/app/stats/command/command.proto#L54l) to set statistics either.

You probably need to migrate data corruption by checking PID and startup time of v2ray when updating your database.
